crates/bevy_app/src/app.rs

```diff
@@ -209,155 +98,160 @@ impl App {
 
     /// Creates a new empty [`App`] with minimal default configuration.
     ///
-    /// This constructor should be used if you wish to provide custom scheduling, exit handling, cleanup, etc.
+    /// Use this constructor if you want to customize scheduling, exit handling, cleanup, etc.
     pub fn empty() -> App {
-        let mut world = World::new();
-        world.init_resource::<Schedules>();
         Self {
-            world,
-            runner: Box::new(run_once),
-            sub_apps: HashMap::default(),
-            plugin_registry: Vec::default(),
-            plugin_name_added: Default::default(),
-            main_schedule_label: Box::new(Main),
-            building_plugin_depth: 0,
+            sub_apps: SubApps {
+                main: SubApp::new(),
+                sub_apps: HashMap::new(),
+            },
```

Comment:
Please look at #9122. I was asked to split this part of the changes into a separate PR.

Reply:
Yeah I understand the motivation behind this PR. I'm specifically asking about these two fields.

Why is it defined like this

```rust
struct App {
    sub_apps: SubApps,
    ...
}
struct SubApps {
    main: SubApp,
    sub_apps: HashMap<AppLabelId, SubApp>,
}
```

instead of like this?

```rust
struct App {
    main: SubApp,
    sub_apps: HashMap<AppLabelId, SubApp>,
    ...
}
```

Reply:
So you don't have to write the contents of `SubApps::update` by hand every time?

Reply:
Would this be better?
```rust
struct App {
    main: SubApp,
    sub_apps: SubApps,
    ...
}

struct SubApps(HashMap<AppLabelId, SubApp>);

impl SubApps {
   fn extract(&mut self, main: &mut SubApp) {
       ...
   }
   
   ...
}

Reply:
Oh that makes sense. If `SubApps` is meant to be used externally, then there should probably be a way of accessing it.

Reply:
Would this be better?
```rust
struct App {
    main: SubApp,
    sub_apps: SubApps,
    ...
}

#[derive(Deref, DerefMut)]
struct SubApps(HashMap<AppLabelId, SubApp>);

impl SubApps {
   fn extract(&mut self, main: &mut SubApp) {
       ...
   }
   
   ...
}

Reply:
Excluding the main app from the sub-apps collection does make more sense to me, yeah.

Reply:
What about pushing things in the other direction and try to make the main app less special?

```rust
struct App { 
    active_app_label: AppLabel, 
    sub_apps: SubApps, ... 
}

impl SubApps {
    fn extract(&mut self, main_label: AppLabel) {
        let main_app = self.0.get(main_label);
        ...
    }
}
```